.github/workflows/rust.yml

```diff
@@ -22,17 +22,44 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build:
+  check:
     # Run on external PRs and pushes to branches on the repo
     # This is to avoid double running on PRs from internal branches
     if: github.event_name == 'push' || github.event_name == 'schedule' || github.event.pull_request.head.repo.full_name != github.repository
+
+    name: Cargo check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
```

Comment:
Shouldn't we do this step with `stable`? We don't "officially" support nightly toolchain (and keep in mind that it might break pretty much any day, despite being generally very stable over the past ~5 years), and passing `nightly` CI jobs aren't even required for merging.

Reply:
I'm fine with using `stable`. I just wanted to unify toolchain use to maximise cache reuse.